src/codegen/spirv.zig

```diff
@@ -1377,11 +1372,12 @@ const NavGen = struct {
     }
 
     fn ptrType2(self: *NavGen, child_ty: Type, storage_class: StorageClass, child_repr: Repr) !IdRef {
+        const target = self.getTarget();
         const key = .{ child_ty.toIntern(), storage_class, child_repr };
         const entry = try self.ptr_types.getOrPut(self.gpa, key);
         if (entry.found_existing) {
             const fwd_id = entry.value_ptr.ty_id;
-            if (!entry.value_ptr.fwd_emitted) {
+            if (target.os.tag == .opencl and !entry.value_ptr.fwd_emitted) {
```

Comment:
These need to be emitted regardless of OS. I guess the main problem here was that `builtin.cpu` emitted a pointer like this? In my recent PR I removed that these would be emitted, so I think this can be reverted without causing problems.

Reply:
yeah forgot to remove these. we also have PSB capability enabled so there's zero reasons to disallow FwdPtrs for vulkan